File: ExceptionsCPP/Source.cpp
//Copyright (c) 2021 RodKingroo

#include <iostream>
#include "BitString.h"

using namespace std;

int main() {
	try {
		string choose_user;
		cout << "What input method do you want to use ? [console / file]: ";
		getline(cin, choose_user);
		if (choose_user == "console") {

			BitString b_str1;
			cout << "Write bit string 1: ";
			cin >> b_str1;
			cout << "Write bit string 2: ";
			string temp;
			getline(cin, temp);
			BitString b_str2(temp);
			BitString b_str_res = b_str1 & b_str2;
			b_str_res.show_result();

		}
		else if (choose_user == "file") {

			BitString b_file1, b_file2;
			b_file1.FileInput("a.txt");
			b_file2.FileInput("b.txt");
			BitString b_file3 = b_file1 & b_file2;
			b_file3.FileOutput();

		}
	}
	catch (runtime_error& error) {
		cout << "Exception: " << error.what() << endl;
	}
	return 0;
}

File: ExceptionsPython/Source.py
#Copyright (c) 2021 RodKingroo

from BitString import MyBitString
import os

try:
    choose_user = str(input("What input method do you want to use?[console/file]: "))
    if choose_user == "console":
        string_1 = MyBitString()
        string_1.input_str()
        string_2 = MyBitString(str(input("Write bit string 2: ")))
        string_3 = string_1 & string_2
        print("Result of conjunction:", end=" ")
        print(string_3)
    if choose_user == "file":
        string_f1 = MyBitString()
        string_f1.filebitInput("input_file1.txt")
        string_f2 = MyBitString()
        string_f2.filebitInput("input_file2.txt")
        string_f_res = string_f1.conjaction(string_f2)
        string_f_res.fileOutput("output_file.txt")
except Exception as error:
    print(error)

os.system("pause")



